### ISSUE TITLE
Confusing "Expected 'fi'" when incorrect command substitution is used

### ISSUE THREAD

**Issue by memotype** [koalaman/shellcheck#1511]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1047, SC1072
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

Not sure if this requires a new check, or fixing the existing checks.

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
wget "http://${FILESERV}/ovs-rpms.txt"
if grep -q "openvswitch-${OVS_VERSION}-.*\.rpm" ovs-rpms.txt; then
  while read rpm; do
    wget -N "http://${FILESERV}/$rpm"
    touch "$(basename $rpm)"
  done <(cat ovs-rpms.txt)
  echo 0
fi
```

Correct code is:

```sh

#!/bin/bash
wget "http://${FILESERV}/ovs-rpms.txt"
if grep -q "openvswitch-${OVS_VERSION}-.*\.rpm" ovs-rpms.txt; then
  while read rpm; do
    wget -N "http://${FILESERV}/$rpm"
    touch "$(basename $rpm)"
  done < <(cat ovs-rpms.txt)
  echo 0
fi
```

(Note the additional `<` character on the "done" line)

#### Here's what shellcheck currently says:

```
  done <(cat ovs-rpms.txt)
       ^-- SC1047: Expected 'fi' matching previously mentioned 'if'.
       ^-- SC1072: Expected 'fi'. Fix any mentioned problems and try again.
```

#### Here's what I wanted or expected to see:

Here's what bash says:

```
bash: syntax error near unexpected token `<(cat ovs-rpms.txt)'
```

Basically, shellcheck knows *something* is wrong, but misidentifies the issue (or doesn't catch it earlier in the parsing). Took me a while to figure out what was wrong, and had to run a simplified version in the shell to figure it out, then looked up the BashFAQ entry to remember how to do the redirect right.

**Comment by e-kwsm** [comment]
Now (0.11.0) shellcheck issues SC1142:

```
In - line 7:
  done <(cat ovs-rpms.txt)
       ^-- SC1047 (error): Expected 'fi' matching previously mentioned 'if'.
       ^-- SC1072 (error): Expected 'fi'. Fix any mentioned problems and try again.
       ^-----------------^ SC1141 (error): Unexpected tokens after compound command. Bad redirection or missing ;/&&/||/|?
       ^-- SC1142 (error): Use 'done < <(cmd)' to redirect from process substitution (currently missing one '<').

```

**Comment by memotype** [comment]
> Now (0.11.0) shellcheck issues SC1142:
> 
> ```
> In - line 7:
>   done <(cat ovs-rpms.txt)
>        ^-- SC1047 (error): Expected 'fi' matching previously mentioned 'if'.
>        ^-- SC1072 (error): Expected 'fi'. Fix any mentioned problems and try again.
>        ^-----------------^ SC1141 (error): Unexpected tokens after compound command. Bad redirection or missing ;/&&/||/|?
>        ^-- SC1142 (error): Use 'done < <(cmd)' to redirect from process substitution (currently missing one '<').
> ```

SEVEN YEARS LATER! Enter: the hero of the scene, e-kwsm.

memotype: "So, I ran into this weird corner case in your shel checker-"

e-kwsm: "FEAR NOT, HUMBLE SHELL SCRIPTER, I WILL SAVE YOU!" *flourishes cape*

memotype: "Thank you e-kwsm! I don't know what I would have done without you!" *swoons*

e-kwsm: "Just doing my job, citizen." *bows*
